ccl/changefeedccl: TestChangefeedNemeses failed

ccl/changefeedccl.TestChangefeedNemeses [failed](https://tanzanite.cluster.engflow.com/invocation/4554492b-d778-4517-93b5-6da66c531a9f) on master @ [2205334339583cea57db33e3c0d992f672e660e8](https://github.com/cockroachdb/cockroach/commits/2205334339583cea57db33e3c0d992f672e660e8):


```
=== RUN   TestChangefeedNemeses
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedNemeses3422207278
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestChangefeedNemeses/pubsub
    helpers_test.go:853: making server as system tenant
    helpers_test.go:928: making pubsub feed factory
    nemeses_test.go:32: random seed: -1732019578594708751
panic: use of Span after Finish. Span: changefntr. Finish previously called at: <stack not captured. Set debugUseAfterFinish>

goroutine 77106 [running]:
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish(0xbd0c900?)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:182 +0x114
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).RecordingType(0xc00a132000)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:442 +0x18
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).IsVerbose(...)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:450
github.com/cockroachdb/cockroach/pkg/util/log.getSpanOrEventLog({0x7b5a370, 0xc00bf59130})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:92 +0x34
github.com/cockroachdb/cockroach/pkg/util/log.Eventf({0x7b5a370, 0xc00bf59130}, {0x6517fbe, 0x2b}, {0xc00572f820, 0x2, 0x2})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:179 +0x5a
github.com/cockroachdb/cockroach/pkg/util/admission.(*WorkQueue).Admit(0xc0043fac00, {0x7b5a370, 0xc00bf59130}, {{0x1}, 0xe2, 0x17919c8c70a7d356, 0x0, 0x2faf080, {0x0, 0x0, ...}})
	github.com/cockroachdb/cockroach/pkg/util/admission/work_queue.go:802 +0x1286
github.com/cockroachdb/cockroach/pkg/util/admission.(*ElasticCPUWorkQueue).Admit(0xc00627de80, {0x7b5a370, 0xc00bf59130}, 0x2faf080, {{0x1}, 0xe2, 0x17919c8c70a7d356, 0x0, 0x2faf080, {0x0, ...}})
	github.com/cockroachdb/cockroach/pkg/util/admission/elastic_cpu_work_queue.go:87 +0xd2
github.com/cockroachdb/cockroach/pkg/util/admission.(*Pacer).Pace(0xc0049a33b0, {0x7b5a370, 0xc00bf59130})
	github.com/cockroachdb/cockroach/pkg/util/admission/pacer.go:43 +0x134
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*batchingSink).runBatchingWorker(0xc00843fb20, {0x7b5a370, 0xc00bf59130})
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/batching_sink.go:402 +0x4d5
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.makeBatchingSink.func1({0x7b5a370?, 0xc00bf59130?})
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/batching_sink.go:514 +0x2a
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.makeBatchingSink.Group.GoCtx.func2()
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x22
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75 +0x56
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 77096
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:72 +0x96
```
<p>Parameters: <code>attempt=1</code>
, <code>run=11</code>
, <code>shard=13</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedNemeses.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32758

--- a/pkg/util/admission/work_queue.go
+++ b/pkg/util/admission/work_queue.go
@@ -799,7 +799,8 @@ func (q *WorkQueue) Admit(ctx context.Context, info WorkInfo) (enabled bool, err
 		q.metrics.incErrored(info.Priority)
 		q.metrics.recordFinishWait(info.Priority, waitDur)
 		deadline, _ := ctx.Deadline()
-		log.Eventf(ctx, "deadline expired, waited in %s queue for %v",
+		// Don't use the cancelled context.
+		log.Eventf(q.ambientCtx, "deadline expired, waited in %s queue for %v",
 			workKindString(q.workKind), waitDur)
 		return true,
 			errors.Newf("work %s deadline expired while waiting: deadline: %v, start: %v, dur: %v",

